### ISSUE TITLE
Weird behavior of #find

### ISSUE THREAD

**Issue by darinpp** [rails/rails#33918]
### Steps to reproduce

Just use any ActiveRecord class (Fund in my case) and pass an empty array to find:
```
pry(main)> ids_to_get = []
[]
pry(main)> result = Fund.find(ids_to_get)
[]
pry(main)> ids_to_get << 45
[
    [0] 45
]
pry(main)> result
[
    [0] 45
]
pry(main)>
```
### Expected behavior
```
pry(main)> ids_to_get = []
[]
pry(main)> result = Fund.find(ids_to_get)
[]
pry(main)> ids_to_get << 45
[
    [0] 45
]
pry(main)> result
[]
pry(main)>
```

### Actual behavior
See above

### System configuration
**5-2-stable**:

**2.4.4**:

**Comment by lucianosousa** [comment]
Same with rails `5.1.6` and ruby `2.3.7`

**Comment by rafaelfranca** [comment]
Closed by e184d1a94e
